### PR TITLE
Fix hvac mode keys

### PR DESCRIPTION
**Pull request by avishorp**
The keys used in the configuration under modbus -> climate -> hvac_mode_register have changed.

## Proposed change

Change the names of the keys used for the configuration of climate mode in Modbus integration. The existing keys use the word "off" which gets parsed as a boolean False instead of a string.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -414,14 +414,20 @@ modbus:
...
-            auto: 0
-            cool: 1
-            heat: 2
-            fan_only: 3
-            dry: 4
+            state_auto: 0
+            state_cool: 1
+            state_heat: 2
+            state_fan_only: 3
+            state_dry: 4
```
Comment (janiversen) [inline source/_integrations/modbus.markdown:421]: I think state_off is missing ?

**Review by janiversen — CHANGES_REQUESTED** [review]

File: source/_integrations/modbus.markdown (hunk 1 above)
Comment (frenck) [inline source/_integrations/modbus.markdown:421]: This is a breaking change. I've left a comment on the parent PR to avoid this.

**Review by frenck — CHANGES_REQUESTED** [review]
This PR is targeting the wrong branch. It targets the `current` branch, but it depends on a change in the Home Assistant codebase that is depending a future release of Home Assistant. This PR should thus target the `next` branch.

Please adjust the target branch and git rebase if needed.

File: source/_integrations/modbus.markdown (hunk 1 above)
Comment (frenck) [reply source/_integrations/modbus.markdown:421]: Nevermind, was explained in the parent PR.

Comment (frenck) [reply source/_integrations/modbus.markdown:421]:
```suggestion
            state_dry: 4
            state_off: 5
```

**Review by janiversen — DISMISSED** [review]
LGTM. apart from the wrong branch, please rebase.

**Comment by frenck** [comment]
Changed base.

**Review by frenck — APPROVED** [review]
Thanks, @avishorp 👍

**Pull request merged**